src/de.hpi.swa.graal.squeak.test/src/de/hpi/swa/graal/squeak/test/AbstractSqueakTestCaseWithImage.java

```diff
@@ -77,6 +76,22 @@ private static void patchImageForTesting() {
             // Patch TestCase>>#performTest, so errors are printed to stderr for debugging purposes.
             patchMethod("TestCase", "performTest", "performTest [self perform: testSelector asSymbol] on: Error do: [:e | e printVerboseOn: FileStream stderr. e signal]");
         }
+        createProcessForTesting(lists);
+    }
+
+    private static void createProcessForTesting(final ArrayObject lists) {
+        final PointersObject newProcess = new PointersObject(image, image.processClass);
+        newProcess.instVarAtPut0Slow(PROCESS.PRIORITY, USER_PRIORITY_LIST_INDEX + 1);
+
+        final PointersObject priority40List = (PointersObject) ArrayObjectReadNode.getUncached().execute(lists, USER_PRIORITY_LIST_INDEX);
+        final Object firstLink = priority40List.instVarAt0Slow(LINKED_LIST.FIRST_LINK);
+        priority40List.instVarAtPut0Slow(LINKED_LIST.FIRST_LINK, newProcess);
+        if (firstLink == NilObject.SINGLETON) {
+            priority40List.instVarAtPut0Slow(LINKED_LIST.LAST_LINK, newProcess);
+        } else {
+            newProcess.instVarAtPut0Slow(PROCESS.NEXT_LINK, firstLink);
+        }
+        newProcess.instVarAtPut0Slow(PROCESS.LIST, priority40List);
```

Comment:
Are you sure the tests run with the wrong priority? The following check suggests otherwise:

```
mx --dy /compiler squeak --code "Processor activeProcess priority" images/trunk64.image
[graalsqueak] Running Squeak/Smalltalk on OpenJDK 64-Bit GraalVM CE 19.2.1 (Graal-compiled)...
[graalsqueak] Image loaded in 1809ms.
Preparing image for headless execution...
[graalsqueak] Evaluating 'Processor activeProcess priority'...
[graalsqueak] Result: 40
```

Reply:
they do run at the right priority initially. But after a timeout, which often happens during the sleep in relinquishProcess, the activeProcess is the idleProcess

Reply:
Ah that's possible! I wonder if it would be easier to just reload the entire image/interpreter in this case? Trying to fix this in the JUnit test seems tedious and error-prone. Or maybe we should fail the build right after a timeout instead. What do you think?

Reply:
Yes, I think failing the build right after a timeout might be the right solution.
Anyway tests are taking too long, and this will not let slip unexplained timeouts which almost always have something bad as a root cause